src/util/replace_symbol.cpp

```diff
@@ -80,7 +81,7 @@ bool replace_symbolt::replace(
     {
       const exprt &e=it->second;
 
-      if(!replace_with_const && e.is_constant())  // Would give non l-value.
+      if(!replace_with_const && !is_lvalue(e))
```

Comment:
This doesn't belong in this commit as this is a non-trivial change,

Previously required `replace_with_const == false` AND `e.id() == ID_constant`. Now requires `replace_with_const == false` AND a super set of the original possible `e`s.

Reply:
Like @thk123 I'm reviewing this patch by patch and yes, there is an issue here.  I'm happy to resolve it by seeing where else in the patch set it would fit rather than moving it but still.

Reply:
This is a separate PR now: #2200.